enable multi-node in external launcher mode

xieyangxu opened the pull request:
Multi-node (nnodes > 1) in external launcher mode is a valid use case.

hmellor reviewed (commented):
Was this an oversight or was it previously unsupported?

If it was previously unsupported, can you link to where support was added?

xieyangxu commented:
> Was this an oversight or was it previously unsupported?
> 
> If it was previously unsupported, can you link to where support was added?

#23691 added multi-node support originally for mp backend but it works for external launcher mode as well.

zhuohan123 reviewed (approved):
Please follow the codex comment to add more asserts and also consider adding some tests.

xieyangxu commented:
/gemini review

xieyangxu commented:
@codex review

xieyangxu commented on vllm/distributed/parallel_state.py:
```diff
@@ -1169,33 +1169,36 @@ def init_distributed_environment(
+                distributed_init_method,
+            )
+        
         distributed_init_method = get_distributed_init_method(ip, port)
```
updated

xieyangxu committed:
enable multi-node in external launcher mode (#29833)

Summary: Pull Request resolved: https://github.com/vllm-project/vllm/pull/29833

Differential Revision: D88115795
